contrib/docker/Dockerfile

```diff
@@ -0,0 +1,98 @@
+##############################################################################
+#                                                                            #
+# *Notice*                                                                   #
+#                                                                            #
+# This Docker image is based on Alpine Linux, which is a `musl` libc-based   #
+# distribution, and not `glibc`-based (e.g., like Ubuntu or Debian).         #
+#                                                                            #
+# To make Boolector work under `musl`, Boolector is "patched" using the      #
+# Windows patch-sets. These patch-sets (predominantly) are used to restrict  #
+# Boolector to the minimum amount of libc that is exposed via MinGW.         #
+#                                                                            #
+# For our purposes, restricting Boolector to the MinGW runtime also means    #
+# that Boolector builds and runs under `musl`.                               #
+#                                                                            #
+# Any references to patching "for Windows" should be read as "patching for   #
+# a minimal libc".                                                           #
+#                                                                            #
+##############################################################################
+
+FROM alpine:3.9
+
+MAINTAINER Andrew V. Jones "andrew.jones@vector.com"
+
+ENV LC_ALL en_US.UTF-8
+ENV LANG en_US.UTF-8
+
+#
+# Deps needed only for build
+#
+ENV BUILD_DEPS \
+    alpine-sdk \
+    cmake \
+    cython \
+    cython-dev \
+    g++ \
+    gcc \
+    git \
+    make \
+    python-dev \
+    sed
+
+#
+# Deps needed for running afterwards
+#
+ENV PERSISTENT_DEPS \
+    bash \
+    libgcc \
+    libstdc++ \
+    python
+
+#
+# Copy our current source tree to the Docker image
+#
```

Comment:
I know you changed this in order to be able to support building local branches. I would still prefer to define `ARG RELEASE=master` and `wget -q https://github.com/boolector/boolector/archive/${RELEASE}.tar.gz -O ...` from the official repository since this will be the more common use case. You can add this command as a comment and add a comment like `If you want to build a local branch of Boolector add these lines and comment out the others...`.

Reply:
Okay, this is done.

I removed the ability to build the local source tree (making this conditional was more effort than it was worth).

You can now provide Docker build arguments of `UPSTREAM` and `BRANCH` to say where to build from (default is `https://github.com/Boolector/boolector/archive` and `master`).

I add this to the documentation to provide an example of how to build from the non-default upstream/branch.